dynawo/sources/Models/CPP/Controls/Voltage/ModelMinMaxMean/DYNModelMinMaxMean.cpp

```diff
@@ -0,0 +1,365 @@
+nbConnectedInputs_(0) {
+}
+
+/**
+ * @brief MinMaxMean model initialization
+ *
+ */
+void
+ModelMinMaxMean::init(const double /*t0*/) {
+  // Nothing to initalize here.
+}
+
+void
+ModelMinMaxMean::initializeFromData(const boost::shared_ptr<DataInterface>& /*data*/) {
+  // not needed
+}
+
+/**
+ * @brief MinMaxMeans model's sizes getter
+ *
+ * Get the sizes of the vectors and matrices used by the solver to simulate
+ * ModelMinMaxMean instance. Used by @p ModelMulti to generate right size matrices
+ * and vector for the solver.
+ */
+void
+ModelMinMaxMean::getSize() {
+  sizeF_ = 0;  // No dynamics
+  sizeY_ = nbConnectedInputs_;  // All voltage inputs and their boolean activity values
+  sizeZ_ = nbConnectedInputs_;
+  sizeG_ = 0;
+  sizeMode_ = 1;
+
+  calculatedVars_.assign(nbCalculatedVars_, 0);
+}
+
+void
+ModelMinMaxMean::evalF(double /*t*/, propertyF_t /*type*/) {
+  // No evalF function deeded
+}
+
+void
+ModelMinMaxMean::evalG(const double /*t*/) {
+  // No root function for this model
+}
+
+void
+ModelMinMaxMean::evalJt(const double /*t*/, const double /*cj*/, SparseMatrix& /*jt*/, const int /*rowOffset*/) {
+  // No evalJt function needed
+}
+
+void
+ModelMinMaxMean::evalJtPrim(const double /*t*/, const double /*cj*/, SparseMatrix& /*jt*/, const int /*rowOffset*/) {
+  // No evalJtPrim function needed
+}
+
+void
+ModelMinMaxMean::evalZ(const double /*t*/) {
+  // No evalZ function needed
+}
+
+void
+ModelMinMaxMean::collectSilentZ(BitMask* /*silentZTable*/) {
+  // Not needed here.
+}
+
+modeChangeType_t
+ModelMinMaxMean::evalMode(const double /*t*/) {
+  return NO_MODE;
+}
+
+void
+ModelMinMaxMean::evalJCalculatedVarI(unsigned /*iCalculatedVar*/, vector<double>& /*res*/) const {
+  // output depends only on discrete variables
+}
+
+void
+ModelMinMaxMean::getIndexesOfVariablesUsedForCalculatedVarI(unsigned int iCalculatedVar, std::vector<int>& indexes) const {
+  // Need to return the variables for the input voltages (and associated booleans ??)
+  switch (iCalculatedVar) {
+    case minValIdx_:
+    case maxValIdx_:
+    case avgValIdx_:
+      for (std::size_t i = 0; i < nbConnectedInputs_; i++) {
+        indexes.push_back(i);  // Adds the voltages
+      }
+      break;
+    default:
+        throw DYNError(Error::MODELER, UndefJCalculatedVarI, iCalculatedVar);
+  }
+}
+
+
+double
+ModelMinMaxMean::evalCalculatedVarI(unsigned iCalculatedVar) const {
+  double out = 0.0f;
+  switch (iCalculatedVar) {
+  case minValIdx_:
+    out = computeMin();
+    break;
+  case maxValIdx_:
+    out = computeMax();
+    break;
+  case avgValIdx_:
+    out = computeMean();
+    break;
+
+  default:
+    throw DYNError(Error::MODELER, UndefCalculatedVarI, iCalculatedVar);  // Macro defined in DYNMacrosMessage
+    break;
+  }
+
+  return out;
+}
+
+void
+ModelMinMaxMean::evalCalculatedVars() {
+  calculatedVars_[minValIdx_] = computeMin();
+  calculatedVars_[maxValIdx_] = computeMax();
+  calculatedVars_[avgValIdx_] = computeMean();
```

Comment:
average or mean?

Reply:
switched to 'average', losing the catchy phrases, but gaining in readability;